gaeguli/fifo-transmit.c

```diff
@@ -360,32 +377,66 @@ _srt_open (SRTInfo * info)
   }
 
   info->sock = srt_socket (AF_INET, SOCK_DGRAM, 0);
+  _apply_socket_options (info->sock);
 
-  if (info->poll_id != SRT_ERROR) {
-    srt_epoll_release (info->poll_id);
+  if (srt_connect (info->sock, sa, sa_len) == SRT_ERROR) {
+    g_debug ("%s", srt_getlasterror_str ());
+    return FALSE;
   }
 
-  info->poll_id = srt_epoll_create ();
+  g_debug ("opened srt socket successfully");
+  return TRUE;
+}
 
-  for (; params->name != NULL; params++) {
-    if (srt_setsockopt (info->sock, 0, params->param, &params->val,
-            sizeof (gint))) {
-      g_error ("%s", srt_getlasterror_str ());
-    }
+static gboolean
+_srt_accept (SRTInfo * info)
+{
+  g_return_val_if_fail (info->sock == SRT_INVALID_SOCK, FALSE);
+  g_return_val_if_fail (info->listen_sock != SRT_INVALID_SOCK, FALSE);
 
+  info->sock = srt_accept (info->listen_sock, NULL, NULL);
+  if (info->sock == SRT_INVALID_SOCK) {
+    return FALSE;
   }
 
-  if (srt_epoll_add_usock (info->poll_id, info->sock, &sock_flags)) {
-    g_warning ("%s", srt_getlasterror_str ());
-    goto failed;
+  srt_close (info->listen_sock);
+  info->listen_sock = SRT_INVALID_SOCK;
+
+  g_debug ("Connection to SRT socket accepted");
+
+  return TRUE;
+}
+
+static gboolean
+_srt_open (SRTInfo * info)
+{
+  g_autoptr (GError) error = NULL;
+
+  g_return_val_if_fail (info->sock == SRT_INVALID_SOCK, FALSE);
```

Comment:
This will cause *ISO C90 forbids mixed declarations and code* warning. It needs to be moved next line of declaration.

Reply:
Does not happen for me. If this is a concern we should set c_std=gnu89 in meson.build.